lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -27,6 +27,8 @@
   bucket
   stateful_set
   cron_job
+  custom_resource_definition
+  custom_resource
```

Comment:
In the initial PR (https://github.com/Shopify/kubernetes-deploy/pull/188/files) this static list was removed in favour of a DAG representation. Do we plan to reintroduce the graph based representation?

Some advantages of the DAG:

- Only expresses ordering between dependencies that actually matter (e.g. multiple solutions to topological sort are valid)
- Can express dependencies between custom resources, a list that considers all CRs of equal deploy priority cannot.